test/micro/org/openjdk/bench/vm/gc/system_gc/AllDead.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2017, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.bench.vm.gc.system_gc;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+
+import java.util.ArrayList;
+import java.util.concurrent.TimeUnit;
+
+@BenchmarkMode(Mode.SingleShotTime)
+@Fork(value=25, jvmArgsAppend={"-Xmx5g", "-Xms5g", "-Xmn3g"})
```

Comment:
It takes ~4.5 min on my box to run all of them, `make CONF=release test TEST='micro:vm.gc.system_gc'`. The variance is rather small, so I wonder this number can be reduced, say 10? (Ofc, this can be easily overwritten on the command line.)

Reply:
On my box (not binding to a single NUMA node) the error is still fairly big for some cases with just 10 forks. For example:
```
Benchmark                       Mode  Cnt    Score    Error  Units
AllLive.gc                        ss    5   61.810 ? 12.743  ms/op
DifferentObjectSizesTreeMap.gc    ss    5  106.660 ? 33.727  ms/op
HalfDeadFirstPart.gc              ss    5   45.340 ?  7.520  ms/op
HalfDeadSecondPart.gc             ss    5   43.968 ?  8.859  ms/op
HalfHashedHalfDead.gc             ss    5  105.144 ? 13.341  ms/op
OneBigObject.gc                   ss    5  114.798 ? 19.266  ms/op
---
Benchmark                       Mode  Cnt    Score    Error  Units
AllLive.gc                        ss   10   61.222 ?  4.838  ms/op
DifferentObjectSizesTreeMap.gc    ss   10  110.614 ? 11.748  ms/op
HalfDeadFirstPart.gc              ss   10   46.089 ?  3.915  ms/op
HalfDeadSecondPart.gc             ss   10   46.085 ?  4.231  ms/op
HalfHashedHalfDead.gc             ss   10  105.468 ?  5.365  ms/op
OneBigObject.gc                   ss   10  118.285 ?  6.044  ms/op
---
Benchmark                       Mode  Cnt    Score   Error  Units
AllLive.gc                        ss   25   59.683 ? 2.747  ms/op
DifferentObjectSizesTreeMap.gc    ss   25  107.748 ? 5.235  ms/op
HalfDeadFirstPart.gc              ss   25   46.860 ? 1.570  ms/op
HalfDeadSecondPart.gc             ss   25   44.730 ? 1.261  ms/op
HalfHashedHalfDead.gc             ss   25  104.233 ? 2.418  ms/op
OneBigObject.gc                   ss   25  114.868 ? 2.398  ms/op
```

Especially for `DifferentObjectSizesTreeMap` the error is quite big even with 25 invocation. Since none of these tests will be run in Mach5 or Perf-aurora without anyone explicitly asking for it I think the additional runtime is worth getting scores with higher confidence.